### ISSUE TITLE
posix srtptime acts weird when a '%z' offset is passed

### ISSUE THREAD

**Issue by yds** [luaposix/luaposix#321]
This test script shows how passing '%z' to srtptime() introduces a discrepancy when converting the  epoch time back to human:
<https://Gist.GitHub.com/yds/c5aaf72e5d766e6a26e30e8b9df84f66>
given EDT as my local timezone, running `strptime.lua 20190322` outputs:

```
2019-03-22 01:00:00 -0400
2019-03-22 00:00:27 -0400
2019-03-22 00:00:00 +0000
2019-03-22 00:00:27 +0000
2019-03-22 04:59:33 +0000
2019-03-22 04:00:00 +0000
2019-03-22 01:00:00 +0000
2019-03-22 00:00:27 +0000
```

I would expect the minutes and seconds to match in every print() -- they do not. why is that?

My test environment is FreeBSD 12-STABLE with [lang/luajit](https://github.com/freebsd/freebsd-ports/tree/master/lang/luajit) and [devel/lua-posix](https://github.com/freebsd/freebsd-ports/tree/master/devel/lua-posix)

**Comment by yds** [comment]
running `env TZ=EST5EDT strptime.lua 20190322` outputs expected results:
```
2019-03-22 01:00:00 -0400
2019-03-22 00:00:00 -0400
2019-03-22 00:00:00 +0000
2019-03-22 00:00:00 +0000
2019-03-22 05:00:00 +0000
2019-03-22 04:00:00 +0000
2019-03-22 01:00:00 +0000
2019-03-22 00:00:00 +0000
```

not having env `TZ` set causes the issue.  `strftime()` not honoring `%z` seems wrong, especially with env `TZ` explicitly set.

**Comment by gvvaughan** [comment]
Do you have some code I use to reproduce this?

Is it a bug in your C library? Please try to reproduce with a small C program that calls strftime directly to verify.

**Comment by yds** [comment]
I reached out to #lua on IRC and with the help of @RhodiumToad we figured out that it was  `WITH_ZONEINFO_LEAPSECONDS_SUPPORT=true` in `/etc/src.conf` on my FreeBSD 12-STABLE build.

The magic clue was the 27 seconds from my first example output above.  27 seconds is how many leap seconds we've had since leapseconds started getting tracked. Rebuilding zoneinfo **without** leapseconds support fixes this issue and many others.

While at it @RhodiumToad created the following patch <https://dpaste.de/tyQP/raw> -- he may come back with a more complete patch since according to him:

> @RhodiumToad: the bug of course is that struct tm might have more than 9 fields (in particular, it typically has a tm_gmtoff field as well)

none of the lua-posix.time functions display the tm_gmtoff in my examples above, only the os.date() output has the correct tm_gmtoff output.  So there's still that.

That pretty much sums it up.  My original issue is solved [the 27 second discrepancy], but some low level bugs were uncovered in the process.
